pkg/topology/builder.go

```diff
@@ -106,13 +106,22 @@ func (b *Builder) evaluateTrafficTarget(res *resources, topology *Topology, tt *
 			Service:       svcKey,
 			TrafficTarget: Key{tt.Name, tt.Namespace},
 		}
-		topology.ServiceTrafficTargets[svcTTKey] = stt
 
 		svc, ok := topology.Services[svcKey]
+		// In the current version of the spec, the traffic will always go to a TrafficSplit. So we don't need to evaluate
+		// Pods if there's already a TrafficSplit on the Service.
+		if ok && len(svc.TrafficSplits) > 0 {
+			b.Logger.Warnf("Service %q already has a TrafficSplit attached, TrafficTarget %q won't be evaluated on this service", svcKey, svcTTKey.TrafficTarget)
```

Comment:
(IMO we should write `will not` instead of `won't` in our logging. We're not having an informal conversation, the logs should stay format. but out of scope of this PR)

Reply:
True, I often forget that :+1: